### ISSUE TITLE
Architecture: bBALN & New Staking Mechanism

### ISSUE THREAD

**Issue by BennyOptions** [balancednetwork/balanced-contracts#308]
Proposal: https://gov.balanced.network/t/baln-token-economics-enhancement-vbaln/161

Note:
If an EOA has locked BALN, and they want to lock more BALN, they have two options. For this example, imagine the EOA has 10 BALN locked with a remaining lock time of 1.24 years. Now imagine they want to lock 100 more BALN:

Option 1: Lock 100 BALN for 1.24 years
Option 2: Lock 100 BALN for longer than 1.24 years, which will reset the lock on the original 10 locked BALN to the new lock. For example, he could lock the 100 BALN for 2 years, which would result in 110 BALN locked for 2 years.

In summary, there is only one timer per EOA.

**Comment by BennyOptions** [comment]
Curve contract equivalent

https://etherscan.io/address/0x5f3b5DfEb7B28CDbD7FAba78963EE202a494e2A2#code
